upup/pkg/fi/cloudup/vsphere/vsphere_cloud.go

```diff
@@ -32,27 +40,50 @@ type VSphereCloud struct {
 	Cluster    string
 	Username   string
 	Password   string
+	Client     *govmomi.Client
 }
 
+const (
+	snapshotName string = "LinkCloneSnapshotPoint"
+	snapshotDesc string = "Snapshot created by kops"
+)
+
 var _ fi.Cloud = &VSphereCloud{}
 
 func (c *VSphereCloud) ProviderID() fi.CloudProviderID {
 	return fi.CloudProviderVSphere
 }
 
 func NewVSphereCloud(spec *kops.ClusterSpec) (*VSphereCloud, error) {
+	glog.V(2).Info("Creating vSphere Cloud...", spec.CloudConfig)
 	server := *spec.CloudConfig.VSphereServer
 	datacenter := *spec.CloudConfig.VSphereDatacenter
 	cluster := *spec.CloudConfig.VSphereResourcePool
+
 	username := os.Getenv("VSPHERE_USERNAME")
 	password := os.Getenv("VSPHERE_PASSWORD")
 	if username == "" || password == "" {
 		return nil, fmt.Errorf("Failed to detect vSphere username and password. Please set env variables: VSPHERE_USERNAME and VSPHERE_PASSWORD accordingly.")
 	}
 
-	c := &VSphereCloud{Server: server, Datacenter: datacenter, Cluster: cluster, Username: username, Password: password}
-	// TODO: create a client of govmomi here?
-	return c, nil
+	u, err := url.Parse(fmt.Sprintf("https://%s:%d/sdk", server, 443))
+	glog.V(2).Info("Creating vSphere Cloud URL is:", u)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
I believe the order of glog.Info and if block should be swapped. No point in logging nil url.

For my knowledge, what purpose is glog.V(2) serving? Some places I see glog.Info being used.